astropy/units/equivalencies.py

```diff
@@ -207,6 +208,12 @@ def doppler_radio(rest):
     <Quantity -31.209092088877583 km / s>
     """
 
+    try:
+        rest.to(si.Hz, spectral())
+    except (AttributeError,UnitsError) as ex:
+        raise UnitsError("The 'rest' value must be a spectral equivalent "
+                         "(frequency, wavelength, or energy).")
```

Comment:
It seems to me that this validation, if it is necessary, could be moved into a single subroutine to be called where appropriate, rather than copied to several places.